src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala

```diff
@@ -275,6 +312,7 @@ class Initializer(config: Config) extends SparkSessionWrapper {
       mapper.readValue[OverwatchParams](overwatchArgs)
     }
 
+    validateApiEnvConfig(rawParams.apiEnvConfig)
```

Comment:
the validation should be done on the object after it's been created, not the rawParams. Please move this down a bit and utilize the constructed object for validation, not the raw values. This is important in case some setter has (or winds up with) logic that adjusts these values as the object[s] get built.

Reply:
done